app/javascript/react/containers/ShopIndexContainer.js

```diff
@@ -31,26 +32,46 @@ class ShopIndexContainer extends Component {
       .catch(error => console.error(`Error in fetch: ${error.message}`))
   }
 
-  render() {
-    let shops = this.state.shops.map((shop) => {
-      return(
-      <ShopIndexTile
-        key={shop.id}
-        id={shop.id}
-        name={shop.name}
-        address={shop.address}
-        city={shop.city}
-        state={shop.state}
-        zip={shop.zip}
-        phoneNumber={shop.phone_number}
-      />)
+  handleShopFilterChange = (e) => {
+    const search = e.target.value
+    this.setState({
+      shopFilter: search
     })
+  }
+
+  filterShops = (shops) => {
+    const { shopFilter } = this.state
+    return shops.filter(shop => shop.name.toLowerCase().indexOf(shopFilter.toLowerCase()) > -1)
+  }
+
+  render() {
+    const { shops, shopFilter } = this.state
+
     return(
       <div>
-        {shops}
+        <div>
+          <input
+            type="text"
+            value={shopFilter}
+            placeholder="Type to filter on name"
+            onChange={this.handleShopFilterChange}
+          />
+        </div>
+        {
+          this.filterShops(shops).map((shop) => <ShopIndexTile
+            key={shop.id}
+            id={shop.id}
+            name={shop.name}
+            address={shop.address}
+            city={shop.city}
+            state={shop.state}
+            zip={shop.zip}
+            phoneNumber={shop.phone_number}
+          />)
+        }
```

Comment:
this should not be inside of your JSX, but instead be done above the return inside of your render method

Reply:
Note: this search bar is more of a filter bar, it just filters the shops already in state. The logic is unique, so if we want to move ahead with this I would love to see it in action on someone’s machine.